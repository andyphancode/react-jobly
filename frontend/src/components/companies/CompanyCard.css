.CompanyCard { 
    text-decoration: none;
    text-align: justify;
}

