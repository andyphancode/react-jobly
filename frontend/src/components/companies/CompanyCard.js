import React from "react";
import { Link } from "react-router-dom";
import "./CompanyCard.css";
import { Card, CardTitle, CardBody, CardText } from "reactstrap";

/**
 * 
 *  Company card to show quick details about company. Rendered for each card in CompanyList.
 * 
 */

function CompanyCard({ handle, name, description, logoUrl }) {
    // logoUrl appears to be invalid as of currently, not sure why
    return (
        <Link className="CompanyCard" to={`/companies/${handle}`}>
            <Card className="my-3">
                <CardBody>
                    <CardTitle tag="h4" className="CardTitle">{name}</CardTitle>
                    <CardText>{description}</CardText>
                </CardBody>
            </Card>
        </Link>
    )
}

export default CompanyCard;