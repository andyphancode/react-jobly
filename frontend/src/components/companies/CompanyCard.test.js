import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard
            handle="random"
            name="Random"
            description="Random company."
            logo_url="https://upload.wikimedia.org/wikipedia/commons/thumb/2/2f/Google_2015_logo.svg/250px-Google_2015_logo.svg.png"
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
