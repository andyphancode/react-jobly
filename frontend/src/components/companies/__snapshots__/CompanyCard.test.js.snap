// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard"
    href="/companies/random"
  >
    <div
      class="my-3 card"
    >
      <div
        class="card-body"
      >
        <h4
          class="CardTitle card-title"
        >
          Random
        </h4>
        <p
          class="card-text"
        >
          Random company.
        </p>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
